### ISSUE TITLE
Zero terminated strings broken since 1.1.3

### ISSUE THREAD

**Issue by RobQuistNL** [keichi/binary-parser#16]
Hiya Keichi,

Thanks a lot for the library! Really sweet. I've just noticed that since 1.1.3 the zero-terminated string stuff is broken.

Code:
```js
var Parser = require('binary-parser').Parser;
var StringOptions = {length: 99, zeroTerminated:true};

// Build an IP packet header Parser
var ipHeader = new Parser()
    .skip(1, StringOptions)
    .string("command", StringOptions)
    .string("username", StringOptions);

// Prepare buffer to parse.
var buf = new Buffer('0e757365726e616d6500526f6200', 'hex');

// Parse buffer and show result
console.log(ipHeader.parse(buf));
```

Output on 1.1.2:
```bash
$ npm install; node test.js
binary-parser@1.1.2 node_modules\binary-parser
{ command: 'username', username: 'Rob' }
```

Output on 1.1.3+
```bash
$ npm install; node test.js
binary-parser@1.1.3 node_modules\binary-parser
{ command: 'username\u0000Rob\u0000', username: '' }
```

Don't have the time to look (yet) but maybe I can check it out lateron.

**Comment by keichi** [comment]
Thanks for reporting, I will look into this...

**Comment by keichi** [comment]
I'm suspecting this commit: 3f2d21266d7fb79a3fcc92eb1d553c4328d2957a ; not sure yet.

**Comment by RobQuistNL** [comment]
Yep, quite sure it was that one. Just weird how the unit tests didn't pick it up - you might want to add this specific case to test it.

**Comment by keichi** [comment]
I fixed the bug, added some unit tests and published a new version to npm (1.1.5) !

**Comment by RobQuistNL** [comment]
Keichi, very nice! Thanks a ton. Good to see you working on this so actively. I've been using this for an online game i'm building.

:+1:
